demo/predictions/js/data.js

```diff
@@ -0,0 +1,128 @@
+class DataExecutionManager {
```

Comment:
**suggestion:** why not naming it `ExecutionData` and remove the `Manager` part.
Manager is always a very generic name and here, we have a data structure only. IMHO, this better communicates the intent and make things easier to read.